### ISSUE TITLE
Bogus file descriptor when communicating with varnishd

### ISSUE THREAD

**Issue by espebra** [varnish/vagent2#106]
From stracing varnish-agent:

    [pid 19689] write(4294967295, "ping", 4) = -1 EBADF (Bad file descriptor)
    [pid 19689] write(2, "Write error CLI socket: Bad file descriptor\n", 44) = 44
    [pid 19689] close(4294967295)           = -1 EBADF (Bad file descriptor)

**Comment by KristianLyng** [comment]
Ok, this is obviously a failed open(). That's not a bogus fd, but -1!  I'll take a look.

**Comment by KristianLyng** [comment]
@espebra 

I've played a bit with this to harden vadmin. I've got a patch that should prevent it from doing that invalid write(), but I'm curious as to what you did to notice this in the first place? It's not like this would crash the agent under normal conditions - just ... not work :D

Basically this could happen any time the agent fails to connect to varnish (The logic is stupid and obviously flawed: "if (not-connected) { connect(...); } write(...);", so regardless of whether the connect succeeds, we try to use the fd). 

Preventing the underlying write() is trivial, but I'm curious if this goes deeper, as there are many calls to cli_write() that go un-checked at the moment.
